lib/client/ticks.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+function prepare (client, opts) {
+  opts = checkOptions(client, opts);
+
+  if (opts.scaleY === 'linear') {
+    return prepareLinear(client, opts);
+  } else {
+    return prepareLog(client, opts);
+  }
+}
+
+function checkOptions (client, opts) {
+  opts = opts || {};
+  opts.scaleY = opts.scaleY || client.settings.scaleY;
+  //assume any values from opts are already scaled
+  //do any other scaling here
+  opts.high = opts.high || Math.round(client.utils.scaleMgdl(client.settings.thresholds.bgHigh));
+  opts.targetTop = opts.targetTop || Math.round(client.utils.scaleMgdl(client.settings.thresholds.bgTargetTop));
+  opts.targetBottom = opts.targetBottom || Math.round(client.utils.scaleMgdl(client.settings.thresholds.bgTargetBottom));
+  opts.low = opts.low || Math.round(client.utils.scaleMgdl(client.settings.thresholds.bgLow));
+
+  return opts;
+}
+
+function prepareLog (client, opts) {
+  if (client.settings.units === 'mmol') {
+    return [
+      2.0
+      , Math.round(client.utils.scaleMgdl(client.settings.thresholds.bgLow))
+      , opts.targetBottom
+      , 6.0
+      , opts.targetTop
+      , opts.high
+      , 22.0
+    ];
+  } else {
+    return [
+      40
+      , opts.low
+      , opts.targetBottom
+      , 120
+      , opts.targetTop
+      , opts.high
+      , 400
+    ];
+  }
+}
+
+function prepareLinear (client, opts) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['opts' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1182413504/issues/source?bid=2452374&fileBranchId=2921533#l50)